pages/a/orchestrating-crate-with-docker-112.html

```diff
@@ -0,0 +1,148 @@
+
+```bash
+$ docker network create -d overlay crate-network
+6p7nxdurhzw2hcwy3bjm2z2wo
+```
+
+Finally we are able to run Crate on top of the Docker cluster! For this, a `docker service` is required - which is created by this command:
+
+```bash
+$ eval $(docker-machine env crate-sw-1) # connect your local docker client to the remote machine
+$ docker service create \
+    --name crate \
+    --network crate-network \
+    --mode global \
+    --endpoint-mode dnsrr \
+    --update-parallelism 1 \
+    --update-delay 60s \
+    --mount type=bind,source=/tmp,target=/data \
+  crate:latest \
+    crate \
+    -Des.discovery.zen.ping.multicast.enabled=false \
+    -Des.discovery.zen.ping.unicast.hosts=crate \
+    -Des.gateway.expected_nodes=3 \
+    -Des.discovery.zen.minimum_master_nodes=2 \
+    -Des.gateway.recover_after_nodes=2 \
+    -Des.network.bind=_eth0:ipv4_
+```
+
+There are several things that need explanation in there:
+
+```bash
+    --name crate #\
+    --network crate-network \
+    --mode global \
+```
+These lines name the cluster 'crate', add the previously created overlay network to the service and set the scaling mode to "global", which means that there will always one container per node. Next the parameters:
+
+```bash
+    --endpoint-mode dnsrr \
+    --update-parallelism 1 \
+    --update-delay 60s \
+    --mount type=bind,source=/tmp,target=/data \
+```
+
+define the endpoint mode to be 'dnsrr' (DNS round robin - the dns entry 'crate' should cycle through the available IP addresses), `update-parallelism` and `update-delay` set how many instances should be updated in parallel and with what delay. Finally there is some volume mapping to ensure Crate is not writing data into the container (*don't use /tmp 's tmpfs for storing data. It will be gone after restarting*).
+
+```bash
+  crate:latest \
+    crate \
+    -Des.discovery.zen.ping.multicast.enabled=false \
+    -Des.discovery.zen.ping.unicast.hosts=crate \
+    -Des.gateway.expected_nodes=3 \
+    -Des.discovery.zen.minimum_master_nodes=2 \
+    -Des.gateway.recover_after_nodes=2 \
+    -Des.network.bind=_eth0:ipv4_
+```
+This part is the same for many Docker deployment of Crate: Multicast is disabled (it's not supported on the overlay network), instead the Docker DNS record 'crate' (the service name) is responsible for the cluster to find the other nodes. Since we are working with a 3-node cluster, the `expected_nodes`, `minimum_master_nodes`, and `recover_after_nodes` are set accordingly. The last line tells Crate to explicitly listen and bind to the IPv4 address of eth0 (usually the crate-network NIC), which is necessary since the auto discovery on the virtual adapters might be wrong.
```

Comment:
do you mean to say ..."the same for ANY Docker..." instead of "many?"
As with the previous comments, maybe a bulleted list here, explaining each note separately? 
Maybe a link to another part of the doc that explains the Node number (min,recover_after, expected) parameters? Why those are 3,2,2 for a 3 node cluster.